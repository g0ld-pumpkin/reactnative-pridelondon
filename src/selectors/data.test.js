// @flow
import {
  generateEvent,
  generateFeaturedEvents,
  generatePerformance,
  sampleArrayOf,
  sampleOne
} from "../data/__test-data";
import type { State as DataState } from "../reducers/data";
import {
  selectEvents,
  selectEventsMap,
  selectEventById,
  selectFeaturedEvents,
  selectFeaturedEventsByTitle,
  selectLoading,
  selectPerformanceById,
  selectPerformancesMap,
  selectRefreshing,
  resolveEvents
} from "./data";

const createData = (): DataState => ({
  events: [],
  featuredEvents: [],
  headerBanners: [],
  images: {},
  paradeGroups: [],
  performances: {},
  sponsors: [],
  amenities: [],
  loading: false,
  refreshing: false,
  noDataReceived: false
});

describe("selectLoading", () => {
  it("selects events", () => {
    const data = createData();
    data.loading = true;
    const selected = selectLoading(data);

    expect(selected).toEqual(true);
  });
});

describe("selectRefreshing", () => {
  it("selects events", () => {
    const data = createData();
    data.refreshing = true;
    const selected = selectRefreshing(data);

    expect(selected).toEqual(true);
  });
});

describe("selectEvents", () => {
  it("selects events", () => {
    const data = createData();
    const selected = selectEvents(data);

    expect(selected).toEqual(data.events);
  });
});

describe("selectEventsMap", () => {
  it("transforms array of events to map of events", () => {
    const selected = selectEventsMap(sampleArrayOf(generateEvent)(3));

    expect(selected).toMatchSnapshot();
  });
});

describe("selectEventById", () => {
  it("selects event property", () => {
    const event = sampleOne(generateEvent, { seed: 398 });
    const eventsById = {
      test: event
    };
    const selected = selectEventById(eventsById, "test");

    expect(selected).toEqual(event);
  });
});

describe("selectPerformancesMap", () => {
  it("selects performances", () => {
    const data = createData();
    const selected = selectPerformancesMap(data);

    expect(selected).toEqual(data.performances);
  });
});

describe("selectPerformanceById", () => {
  it("selects performance", () => {
    const performance = sampleOne(generatePerformance, { seed: 398 });
    const performancesById = {
      test: performance
    };
    const selected = selectPerformanceById(performancesById, "test");

    expect(selected).toEqual(performance);
  });
});

describe("selectFeaturedEvents", () => {
  it("selectsFeaturedEvents", () => {
    const data = createData();
    const selected = selectFeaturedEvents(data);

    expect(selected).toEqual(data.featuredEvents);
  });
});

describe("selectFeaturedEventsByTitle", () => {
  it("returns nothing when no featured events with the specified title exist", () => {
    const featuredEventsList = sampleArrayOf(generateFeaturedEvents)(3);
    const events = selectFeaturedEventsByTitle(featuredEventsList, "missing");
    expect(events).toEqual(undefined);
  });

  it("returns FeaturedEvents with the title", () => {
    const featuredEventsList = sampleArrayOf(generateFeaturedEvents)(3);
    const events = selectFeaturedEventsByTitle(
      featuredEventsList,
      featuredEventsList[1].fields.title
    );
    expect(events).toEqual(featuredEventsList[1]);
  });
});

describe("resolveEvents", () => {
  it("resolves array of FieldRefs to Events", () => {
    const eventA = sampleOne(generateEvent, { seed: 1236 });
    const eventB = sampleOne(generateEvent, { seed: 2236 });
    const eventC = sampleOne(generateEvent, { seed: 4236 });
    const eventMap = {
      [eventA.id]: eventA,
      [eventB.id]: eventB,
      [eventC.id]: eventC
    };
    const references = [{ sys: { id: eventA.id } }, { sys: { id: eventB.id } }];
    const resolved = resolveEvents(eventMap, references);
    expect(resolved).toEqual([eventA, eventB]);
  });

  it("omits any unresolvable FieldRefs", () => {
    const eventA = sampleOne(generateEvent, { seed: 1236 });
    const eventB = sampleOne(generateEvent, { seed: 2236 });
    const eventC = sampleOne(generateEvent, { seed: 4236 });
    const eventMap = {
      [eventA.id]: eventA,
      [eventB.id]: eventB,
      [eventC.id]: eventC
    };
    const references = [{ sys: { id: "missing" } }];
    const resolved = resolveEvents(eventMap, references);
    expect(resolved).toEqual([]);
  });
});
