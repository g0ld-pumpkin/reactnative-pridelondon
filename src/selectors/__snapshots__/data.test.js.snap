// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectEventsMap transforms array of events to map of events 1`] = `
Object {
  "5obj688Cj3vM": Object {
    "contentType": "event",
    "fields": Object {
      "accessibilityDetails": "accessibilityDetails",
      "accessibilityOptions": Array [
        "0hWS71X6v20dL18",
        "02X4sXo8lGzGTL8928CyO",
        "t563ZL67SSGD0K1un2owA0C5",
        "VSH6Qo8k342G5rZP",
      ],
      "addressLine1": "addressLine1",
      "addressLine2": "addressLine2",
      "audience": Array [
        "iTSBsb1G99KHmZFVYS4fQAl1S741",
        "3c9BUjc848u71VcYRCWX0",
        "7N5VV4l02bp2gOaA",
        "p7IW7MI99tBNamFvDd8Y64ezv5",
      ],
      "city": "city",
      "email": "email",
      "endTime": "2018-07-07T03:00+00:00",
      "eventCategories": Array [
        "Cabaret and Variety",
        "Health",
        "Nightlife",
        "Music",
        "Talks and Debates",
      ],
      "eventDescription": "eventDescription",
      "eventPriceHigh": 10,
      "eventPriceLow": 0,
      "eventsListPicture": Object {
        "sys": Object {
          "id": "b8Gmbw5l8QQWmULc94kd5U4",
        },
      },
      "individualEventPicture": Object {
        "sys": Object {
          "id": "0F46iOVTNxqz9NJl25pl3",
        },
      },
      "location": Object {
        "lat": 0,
        "lon": 10,
      },
      "locationName": "locationName",
      "name": "name",
      "performances": Array [
        Object {
          "sys": Object {
            "id": "BAPd",
          },
        },
        Object {
          "sys": Object {
            "id": "qC9h3pCZd6",
          },
        },
      ],
      "phone": "phone",
      "postcode": "postcode",
      "recurrenceDates": Array [
        "04/07/2018",
      ],
      "stage": true,
      "startTime": "2018-07-07T00:00+00:00",
      "ticketingUrl": "ticketingUrl",
      "venueDetails": Array [
        "Fos6Vj70ntMWl",
        "8t85",
      ],
    },
    "id": "5obj688Cj3vM",
    "locale": "en-GB",
    "revision": 1,
  },
  "G36Nw4N4Qb1HE305M0V": Object {
    "contentType": "event",
    "fields": Object {
      "accessibilityDetails": "accessibilityDetails",
      "accessibilityOptions": Array [
        "xipaGqs0am",
        "b4515eG8Auw",
        "LHLlMyK1nhZW762jVQAK1n704rrk3",
        "eC36IkPHrjr7o5l4XuI6QBCe",
      ],
      "addressLine1": "addressLine1",
      "addressLine2": "addressLine2",
      "audience": Array [
        "5id",
        "OnT4h29Nlhm2ECm2l",
        "YvTD0M251aE83S",
        "nR658",
        "JUc66PnZ6p967v7Xc",
      ],
      "city": "city",
      "email": "email",
      "endTime": "2018-07-07T03:00+00:00",
      "eventCategories": Array [
        "Community",
      ],
      "eventDescription": "eventDescription",
      "eventPriceHigh": 10,
      "eventPriceLow": 0,
      "eventsListPicture": Object {
        "sys": Object {
          "id": "z2tvI2H52F4XG",
        },
      },
      "individualEventPicture": Object {
        "sys": Object {
          "id": "Buka8JAKLif9k6d",
        },
      },
      "location": Object {
        "lat": 0,
        "lon": 10,
      },
      "locationName": "locationName",
      "name": "name",
      "performances": Array [
        Object {
          "sys": Object {
            "id": "0qHqaN4rV6suzpo4F3aF5v4vb5G",
          },
        },
        Object {
          "sys": Object {
            "id": "0lVlx1I6219T2410KM4v4E1",
          },
        },
        Object {
          "sys": Object {
            "id": "92P16UVWt7hWEDGm97XT7Fb4I",
          },
        },
      ],
      "phone": "phone",
      "postcode": "postcode",
      "recurrenceDates": Array [
        "18/08/2018",
        "18/08/2018",
      ],
      "stage": true,
      "startTime": "2018-07-07T00:00+00:00",
      "ticketingUrl": "ticketingUrl",
      "venueDetails": Array [
        "mn204uE5Cedi2",
        "7rIC1",
        "869bRJ27f7",
      ],
    },
    "id": "G36Nw4N4Qb1HE305M0V",
    "locale": "en-GB",
    "revision": 1,
  },
  "hG": Object {
    "contentType": "event",
    "fields": Object {
      "accessibilityDetails": "accessibilityDetails",
      "accessibilityOptions": Array [
        "v4817p72g9B28",
        "H2Cbs5bqh",
        "Wagc8Wv8jK6K6D77490GB",
        "YIn7xDxQbT",
      ],
      "addressLine1": "addressLine1",
      "addressLine2": "addressLine2",
      "audience": Array [
        "1fx905L",
        "9OgXUt",
        "46giMKK1fEEP9IsCd57b46Lez9W75b",
        "hE6X046ClAwqcJIqU4Jx14eaBPNpeh",
      ],
      "city": "city",
      "email": "email",
      "endTime": "2018-07-07T03:00+00:00",
      "eventCategories": Array [
        "Social and Networking",
        "Talks and Debates",
      ],
      "eventDescription": "eventDescription",
      "eventPriceHigh": 10,
      "eventPriceLow": 0,
      "eventsListPicture": Object {
        "sys": Object {
          "id": "8iB636ntgq3B632hFy6uHIt7kC",
        },
      },
      "individualEventPicture": Object {
        "sys": Object {
          "id": "e6VCvM1t1S91yP87Jb4Xy6ZCK",
        },
      },
      "location": Object {
        "lat": 0,
        "lon": 10,
      },
      "locationName": "locationName",
      "name": "name",
      "performances": Array [
        Object {
          "sys": Object {
            "id": "B701ApOpyJl",
          },
        },
      ],
      "phone": "phone",
      "postcode": "postcode",
      "recurrenceDates": Array [
        "12/07/2018",
        "12/07/2018",
        "13/08/2018",
        "07/07/2018",
      ],
      "stage": true,
      "startTime": "2018-07-07T00:00+00:00",
      "ticketingUrl": "ticketingUrl",
      "venueDetails": Array [
        "C68l",
        "M6cCc5X3p60J55sCfBCs7jD8l",
        "4Ll0RYyIa2y8dZ3FCBY6ES82lvQ4P",
      ],
    },
    "id": "hG",
    "locale": "en-GB",
    "revision": 1,
  },
}
`;
